### PR TITLE
NAS-119204 / 23.10 / Allow setting failover network interface options

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 2:00

**Review by yocalebo — CHANGES_REQUESTED** [review]
1. `ipv4_dhcp`, `ipv6_dhcp` are not supported on HA systems
2. `failover_vhid` is no longer required and ignored on SCALE HA
3.

File: midcli/gui/network/interface/create_update.py [hunk 1]
```diff
@@ -20,15 +20,31 @@ class NetworkInterfaceCreate(Steps):
...
+
+        if self.method == StepsMethod.CREATE:
+            result.append(Input("type"))
+
+        result.extend([
             Input("name"),
             Input("description"),
             Input("ipv4_dhcp"),
             Input("ipv6_auto"),
             Input("aliases", delegate=AliasesInputDelegate),
```
Comment (yocalebo) [inline midcli/gui/network/interface/create_update.py:33]: This doesn't seem to enforce or handle the fact if a user provides an ip address without netmask information?

**Comment by themylogin** [comment]
time 1:00

**Review by yocalebo — APPROVED** [review]

**Pull request merged**